scripts/lib/website-comparisons.ts

```diff
@@ -0,0 +1,46 @@
+import * as cheerio from 'cheerio';
+import { createHash } from 'node:crypto';
+
+// Return and clean the HTML body portions of a returned website.
+export function cleanWebsiteData(response: string): string {
+  const $ = cheerio.load(response);
+  // Only check for the HTML body to try and avoid metadata and scripts that may change per-system/call.
+  if ($('body').length > 0) {
+    return $('body').contents().text().trim();
+  }
+  // If the website does not have a body tag, return all found html (for now).
+  return $.html().trim();
+}
+
+// Check whether a link is for a pdf or a website page.
+export function isPdf(link: string, response: string): boolean {
+  if (link.endsWith('.pdf')) {
+    return true;
+  }
+  // It is still possible a link is to a PDF, even if the URL doesn't end with pdf. For these, we check the object type.
+  const $ = cheerio.load(response);
+  if ($('body').length > 0) {
+    if ($('[type="application/pdf"]').length > 0) {
+      return true;
+    }
+  }
+  // Otherwise, assume this is a normal website.
+  return false;
+}
+
+// Given a website link and its html, return a hash of the cleaned website. Do not return a hash if the site is for a PDF.
+export function checkWebsiteAndReturnHash(
+  link: string,
+  response: string,
+): string | null {
+  if (isPdf(link, response)) {
+    return null;
```

Comment:
Out of curiosity, why not just return a hash of the entire PDF content?

This could be naive, but I would guess PDFs change rarely. Doesn't have to happen in this PR either way.

Reply:
This is what I'll be aiming to do in a further CL; the issue is that I'm unable to get the PDF content via Cheerio's API - I need to use a PDF parsing library, which has some complexities I'd rather deal with separately.